sa/precertificates.go

```diff
@@ -61,6 +61,15 @@ func (ssa *SQLStorageAuthority) AddPrecertificate(ctx context.Context, req *sapb
 	}
 
 	_, overallError := db.WithTransaction(ctx, ssa.dbMap, func(txWithCtx db.Executor) (interface{}, error) {
+		// Select to see if precert exists
+		var rows []interface{}
+		results, err := txWithCtx.Select(&rows, "SELECT id FROM precertificates WHERE serial=?", serialHex)
```

Comment:
I think it's simpler (and maybe faster?) to do:
```suggestion
		c, err := txWithCtx.SelectInt("SELECT count(id) FROM precertificates WHERE serial=?", serialHex)
```
and then compare `c > 0` instead of `len(results) > 0`.

Reply:
From the gorp docs:

>SelectInt executes the given query, which should be a SELECT statement for a single integer column, and returns the value of the first row returned. If no rows are found, zero is returned.

So using 0 as a sentinel would be `c  ==  0`.

Alternatively, you can keep the `SelectOne()` and your sentinel would be: `errors.Is(err, sql.ErrNoRows)`

If you keep the select one, make a composite type with an exported field of `ID int64` and pass a slice of that composite type as the holder.

Reply:
Note that my suggestion is to select `count(id)`, not just `id`, so we're not looking for a sentinel here -- we're literally just looking for "how many rows would you have returned?", and if the answer is anything greater than zero, then we have a problem.

Reply:
> Note that my suggestion is to select `count(id)`, not just `id`, so we're not looking for a sentinel here -- we're literally just looking for "how many rows would you have returned?", and if the answer is anything greater than zero, then we have a problem.

Ah, I missed that.

Reply:
I agree with @aarongable's proposal in concept, but there's a catch: we use the `db.Executor` interface here, which is a subset of the methods generally available on `gorp.DbMap`. It includes `Select` and `SelectOne`, but it doesn't include `SelectInt`. I think it's probably not worth plumbing through `SelectInt` (and making sure it's implemented in all relevant DB mocks) for this small optimization. What do you think, @aarongable ?

Reply:
Gah, you're probably right that it isn't worthwhile "just" for this, but I do see a bunch of other places where we could use it. In particular, there are a bunch of places where we use `SelectOne(&count, ...)` in order to get a single integer out of the database. Which is also what we should do here, if we don't plumb `SelectInt` all the way through the wrapper.